pkg/js/helpers.js

```diff
@@ -1879,15 +1879,14 @@ function M365_BUILDER(name, value) {
 
     var r = [];
 
+    // MX host (default: "mail.protection.outlook.com")
+    if (!value.mxHost) {
+        value.mx = 'mail.protection.outlook.com';
+    }
+
     // MX (default: true)
     if (value.mx) {
-        r.push(
-            MX(
-                value.label,
-                0,
-                value.domainGUID + '.mail.protection.outlook.com.'
-            )
-        );
+        r.push(MX(value.label, 0, value.domainGUID + '.' + value.mxHost + '.'));
```

Comment:
The [PowerShell(?) commands](https://learn.microsoft.com/en-gb/purview/how-smtp-dane-works#inbound-smtp-dane-with-dnssec:~:text=Success-,contosotest%2Dcom.o%2Dv1.mx.microsoft,-The%20success%20response) (link to highlight, might not work in all browsers) returns the whole MX hostname: `contosotest-com.o-v1.mx.microsoft`.

Therefore, and to make the helper more flexible in general, I advise to do one of the following:

1. Document how to disable the MX and set the Microsoft-provided hostname. (**preferred**)
2. Allow any MX to be set, possibly multiple MXs

Documentation for the first option could look like this:

```
D("example.com", REG_MY_PROVIDER, DnsProvider(DSP_MY_PROVIDER),
  M365_BUILDER("example.com", {
      mx: false,
      …
  }),
  MX('@', 0, 'example-com.<random>.mx.microsoft.') // Note the trailing dot.
);
```